streaming/dataset.py

```diff
@@ -459,12 +467,11 @@ def __init__(self,
         self.sample_offset_per_stream = np.zeros(self.num_streams, np.int64)
         self.samples_per_stream = np.zeros(self.num_streams, np.int64)
         for stream_id, stream in enumerate(self.streams):
-            stream_shards = stream.get_shards(world, self.allow_unsafe_types)
+            stream.await_index()
```

Comment:
I think this answers my above question, where this is a blocking call to wait for the `index.json` to cache locally.

Reply:
This is on the critical path, loading many shards takes a long time, we need to get started ASAP, hence this way

There’s a faster, slightly more complicated way to do this in the PR desc but might save for future PR